### ISSUE TITLE
[GSdx HW/SW] TSTTB: Corrupted Garbage/Flickering during Certain Sequences.

### ISSUE THREAD

**Issue by MrCK1** [PCSX2/pcsx2#1929]
***PCSX2 version:***
v1.5.0-2059

***PCSX2 options:***
Defaults.

***Plugins used:***
GSdx, issue affects all SW renderers.

***Description of the issue:***
When using any SW renderer, the screen will flicker and becomes filled with garbage during certain sequences of the game.

***How to reproduce the issue:***
Go to any sequence in the game where the professor is talking to the player. Switch to SW mode and the screen will have visible corruption and depth issues.

***Last known version to work:***
Bug was introduced between 0.9.2 and 0.9.4. Compared to the picture below, there was less corruption in 0.9.4 (only the upper green portion used to be corrupt IIRC). [I uploaded a new branch to show the diff](https://github.com/MrCK1/pcsx2/commit/12005a3eba934e859852bf99d394385c3ac3d9c1) for the GS.cpp and GS.h (Most of the changes seem to be related to the GIF, so I'm using this as a starting point and can upload more files if they need to be looked at). 

***PC specifications:***
Intel i5-6600K, GTX 1070, Windows 10 64-bit

GS Dump: [TSTTB_SWGarbage.zip](https://github.com/PCSX2/pcsx2/files/975292/TSTTB_SWGarbage.zip)

![tsttb_swgarbage](https://cloud.githubusercontent.com/assets/16314399/25691941/3dfe8878-306d-11e7-9f46-ae9b32c10671.png)

**Comment by gregory38** [comment]
It could be also related to GIF management on the GS side.

```
Bug was introduced between 0.9.2 and 0.9.4. Compared to the picture below, there was less corruption in 0.9.4 (only the upper green portion used to be corrupt IIRC)
```
So 0.9.2 was good. 0.9.4 breaks partially stuff, and latest master is completely broken. Is it right ?

**Comment by MrCK1** [comment]
Yes that's correct. I'll get more screenshots later from the older versions and look for any more bad commits.

**Comment by MrCK1** [comment]
@gregory38 I didn't find anything significant, but I was wondering if these STQ float values are normal or not. Thanks 😉 

STQ values from corrupted scene.
![stqfloatvalueststtb](https://cloud.githubusercontent.com/assets/16314399/25733825/eb85c57a-312a-11e7-93e3-997476ecc17c.png)

STQ values during FMV that plays after corrupted scene.

![stqfloatvalueststtb_fmv](https://cloud.githubusercontent.com/assets/16314399/25733864/3d0368da-312b-11e7-954e-e281b067223a.png)

**Comment by gregory38** [comment]
If game use integer coordinate, the values are UV not STQ. It depends on the fst bit of prim register

**Comment by ssakash** [comment]
Also Q is from RGBAQ register, not the ST register. I wonder why the GSDumpGUI displays them under the ST register prefix.

**Comment by MrCK1** [comment]
@gregory38 in both cases the FST is the STQ value if that still means anything.

**Comment by gregory38** [comment]
I don't know the GUI but what I mean is potentially the rendering doesn't use those STQ values. So even random value will be acceptable. However,  maybe the issue is the sign of STQ, normally they must share the same sign.

**Comment by MrCK1** [comment]
I forgot to mention that the issue is specific to the GSdx plugin being used. Any version of GSdx after 0.9.2 contains the issue.

**Comment by MrCK1** [comment]
Can someone mark this issue as "GSdx: Texture Cache"? I'm positive this is the cause after reviewing some dumps and finding related instances where striped textures are present.

**Comment by MrCK1** [comment]
Here's a raw SW dump for anybody who wants to take a peek. I haven't been able to figure this one out.

[TSTTB_swcorruption_raw.zip](https://github.com/PCSX2/pcsx2/files/1380303/TSTTB_swcorruption_raw.zip)

**Comment by MrCK1** [comment]
I decided to take another look at this issue.

I made a new GS dump again. It seems that the corruption is correlated with a specific context/screen (not sure if that is technically "correct) - I believe the status has changed slightly in SW, but since no changes have been made recently; I probably just didn't notice it before.

This is what is seen in the first 4 VSyncs of the GS Dump, this pattern will repeat. HW is on the left, SW on the right.

The picture in the 1st VSync isn't supposed to happen at all (on either renderer) and doesn't occur on real hardware. 

The only theory I have now is that something is going wrong related to the scissoring (possibly alluding to the garbage data produced in SW?) and/or context switching - maybe there is a delay on the real hardware? These are just my current thoughts, but if you have any ideas, let me know :P

![sw1](https://user-images.githubusercontent.com/16314399/35207175-87cabe42-ff0f-11e7-9056-6900585b3a62.png)

![sw2](https://user-images.githubusercontent.com/16314399/35207298-2a592b80-ff10-11e7-8f6a-2f390df0403b.png)
![sw3](https://user-images.githubusercontent.com/16314399/35207300-2bcbb604-ff10-11e7-80cd-e24d0e55c83d.png)

![sw4](https://user-images.githubusercontent.com/16314399/35207116-304f15d2-ff0f-11e7-9f33-703bb07492e1.png)
